buzsaki_lab_to_nwb/huszar_hippocampus_dynamics/converter.py

```diff
@@ -0,0 +1,179 @@
+                self.sorting_extractor.set_property(
+                    ids=unit_ids,
+                    key="cell_type",
+                    values=[str(celltype_mapping[str(x[0])]) for x in celltype_info["label"][0][0][0]],
+                )
+
+    def get_metadata(self) -> dict:
+        metadata = super().get_metadata()
+        session_path = Path(self.source_data["file_path"]).parent
+        session_id = session_path.stem
+        # TODO: add condition for retrieving ecephys metadata if no recording or lfp are included in conversion
+        metadata["NWBFile"].update(session_id=session_id)
+
+        unit_properties = []
+        cellinfo_file_path = session_path / f"{session_id}.spikes.cellinfo.mat"
+        if cellinfo_file_path.is_file():
+            cell_info_fields = self.cell_info_fields
+            if "cluID" in cell_info_fields:
+                unit_properties.append(
+                    dict(
+                        name="clu_id",
+                        description="0-indexed id of cluster identified from the shank.",
+                    )
+                )
+            if "shankID" in cell_info_fields:
+                unit_properties.append(
+                    dict(
+                        name="group_id",
+                        description="The electrode group ID that each unit was identified by.",
+                    )
+                )
+            if "region" in cell_info_fields:
+                unit_properties.append(
+                    dict(
+                        name="location",
+                        description="Brain region where each unit was detected.",
+                    )
+                )
+        celltype_filepath = session_path / f"{session_id}.CellClass.cellinfo.mat"
+        if celltype_filepath.is_file():
+            celltype_info = scipy.io.loadmat(celltype_filepath).get("CellClass", np.empty(0))
+            if "label" in celltype_info.dtype.names:
+                unit_properties.append(
+                    dict(
+                        name="cell_type",
+                        description="Type of cell this has been classified as.",
+                    )
+                )
+        metadata.update(Ecephys=dict(UnitProperties=unit_properties))
+        return metadata
+
+
+class HuzsarNWBConverter(NWBConverter):
+    """Primary conversion class for the Huzsar hippocampus data set."""
+
+    data_interface_classes = dict(
+        Behavior=HuzsarBehaviorInterface,
+        Sorting=CellExplorerSortingInterface,
+    )
+
+    def __init__(self, source_data: dict, verbose: bool = True):
+        super().__init__(source_data=source_data, verbose=verbose)
+
+    def get_metadata(self):
+        # metadata = super().get_metadata()
+        # metadata["NWBFile"].update(session_start_time=session_start, session_id=session_id)
+        # metadata.update(Subject=dict(subject_id=subject_id))
+
+        metadata = super().get_metadata()
+        import datetime as dt
```

Comment:
Lol, thanks for that.

IDK, my controversial personal opinion is that only `numpy` gets the universal distinction and recognition of `np` for short because literally everyone does it

`pandas` and `datetime`, not quite used frequently or widely enough to merit it. Not that I even strictly like `np` for short; aliasing ought only be used for avoiding namespace collisions as anything else harms readability of code and can make it harder for novices to understand Python import structures (I've seen first-timers try doing things like `import np` for example because they saw code that did `np.array(...)`)

Reply:
An example of namespace collisions being something like

```python
from dandi import validate as dandi_validate
from pynwb import validate as pynwb_validate
```

Reply:
>  Python import structures (I've seen first-timers try doing things like import np for example because they saw code that did np.array(...))

Damn! [the curse of knowledge](https://en.wikipedia.org/wiki/Curse_of_knowledge)

Interesting. Yeah, I think that's the problem of having unspecific names and variables exposed in the high level API. `validate` `calculate` `open`, `do_magic_stuff` x P